sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystQl.scala

```diff
@@ -393,8 +393,8 @@ https://cwiki.apache.org/confluence/display/Hive/Enhanced+Aggregation%2C+Cube%2C
             overwrite)
       }
 
-      // If there are multiple INSERTS just UNION them together into on query.
-      val query = queries.reduceLeft(Union)
+      // If there are multiple INSERTS just UNION them together into one query.
+      val query = if (queries.length == 1) queries.head else Union(queries)
```

Comment:
Minor: you might actually just make `Union` smart enough to return the node itself when there is only one.